src/data/matrix/basic.lean

```diff
@@ -26,24 +26,63 @@ open dmatrix
 def matrix (m : Type u) (n : Type u') [fintype m] [fintype n] (α : Type v) : Type (max u u' v) :=
 m → n → α
 
-variables {l m n o : Type*} [fintype l] [fintype m] [fintype n] [fintype o]
+variables {l m n o: Type*} [fintype l] [fintype m] [fintype n] [fintype o]
 variables {m' : o → Type*} [∀ i, fintype (m' i)]
 variables {n' : o → Type*} [∀ i, fintype (n' i)]
 variables {R : Type*} {S : Type*} {α : Type v} {β : Type w}
 
+/-- `equiv.perm.to_matrix` converts a permutation `σ : equiv.perm m` to a matrix
+    containing ones and zeros. This definition is then used to define the Prop `matrix.is_perm`.
+    `equiv.perm.to_prequiv_to_matrix_eq_to_matrix` proves that
+    `σ.to_pequiv.to_matrix = σ.to_matrix` in `pequiv.lean`.-/
+def equiv.perm.to_matrix [decidable_eq m] [has_zero α] [has_one α] (σ : equiv.perm m) :
+matrix m m α
+| i j := if σ i = j then 1 else 0
+
 namespace matrix
 
+instance [inhabited α] : inhabited (matrix m n α) := pi.inhabited _
```

Comment:
What was the motivation for moving these lines?

Reply:
I just "feel" the current order looks better: declare instances first, do add more defns and lemmas

Reply:
When you make this type of change, it's good to do at least one of:

* Making the change in a standalone PR. It's easier for reviewers if they know that all you did was change the order of code
* Add a brief explanation of _why_ you moved things around in the PR docstring.